src/db/userstore.ts

```diff
@@ -0,0 +1,154 @@
+import { IDatabaseConnector } from "./connector";
+import * as uuid from "uuid/v4";
+import { Log } from "../log";
+
+/**
+ * A UserStore compatible with
+ * https://github.com/matrix-org/matrix-appservice-bridge/blob/master/lib/components/user-bridge-store.js
+ * that accesses the database instead.
+ */
+
+const ENTRY_CACHE_LIMETIME = 30000;
+
+export class RemoteUser {
+    public displayname: string|null = null;
+    public avatarurl: string|null = null;
+    public avatarurlMxc: string|null = null;
+    public guildNicks: Map<string, string> = new Map();
+    constructor(public readonly id: string) {
+
+    }
+}
+
+const log = new Log("DbUserStore");
+
+export interface IUserStoreEntry {
+    id: string;
+    matrix: string|null;
+    remote: RemoteUser|null;
+}
+
+export class DbUserStore {
+    private remoteUserCache: Map<string, {e: RemoteUser, ts: number}>;
+
+    constructor(private db: IDatabaseConnector) {
+        this.remoteUserCache = new Map();
+    }
+
+    public async getRemoteUser(remoteId: string): Promise<RemoteUser|null> {
+        const cached = this.remoteUserCache.get(remoteId);
+        if (cached && cached.ts + ENTRY_CACHE_LIMETIME > Date.now()) {
+            return cached.e;
+        }
+        const row = await this.db.Get(
+            "SELECT * FROM user_entries WHERE remote_id = $id", {id: remoteId},
+        );
+        if (!row) {
+            return null;
+        }
+        const remoteUser = new RemoteUser(remoteId);
+        const data = await this.db.Get(
+            "SELECT * FROM remote_user_data WHERE remote_id = $remoteId",
+            {remoteId},
+        );
+        if (data) {
+            remoteUser.avatarurl = data.avatarurl as string|null;
```

Comment:
Maybe also return `null` if no remote data is found? Same argumentation as with channel store